cms/static/js/views/manage_users_and_roles.js

```diff
@@ -0,0 +1,306 @@
+                    click: function(view) { view.hide(); $('#user-email-input').focus(); }
+                }
+            }
+        });
+    }
+
+    function makeAlreadyMemberMessage(messages, email, containerName) {
+        return new PromptView.Warning({
+            title: messages.alreadyMember.title,
+            message: _.template(
+                messages.alreadyMember.messageTpl,
+                {email: email, container: containerName},
+                {interpolate: /\{(.+?)}/g}
+            ),
+            actions: {
+                primary: {
+                    text: messages.alreadyMember.primaryAction,
+                    click: function (view) {
+                        view.hide();
+                        $('#user-email-input').focus();
+                    }
+                }
+            }
+        });
+    }
+
+    function makeChangeRoleErrorMessage(messages, title, message, onErrorCallback) {
+        return new PromptView.Error({
+            title: title,
+            message: message,
+            actions: {
+                primary: {
+                    text: messages.defaults.confirmation,
+                    click: function(view) { view.hide(); onErrorCallback(); }
+                }
+            }
+        });
+    }
+
+    function getEmail(button){
+        return $(button).closest('li[data-email]').data('email');
+    }
+
+    var ManageUsersAndRoles = BaseView.extend({
+        events: function() {
+            var baseEvents = {
+                'click .create-user-button': "addUserHandler",
+                'submit #create-user-form': "createUserFormSubmit",
+                'click .action-cancel': "cancelEditHandler",
+                'keyup': "keyUpHandler",
+                'click .remove-user': "removeUserHandler"
+            };
+            var roleEvents = {};
+            var self = this;
+            for (var i=0; i<self.roles.length; i++) {
+                var role_name = self.roles[i].key;
+                var role_selector = 'click .user-actions .make-'+role_name;
+
+                (function (role) {
+                    roleEvents[role_selector] = function(event) {
+                        self.handleRoleButtonClick(event.target, role);
+                    };
+                })(role_name);
+            }
+            return _.extend(baseEvents, roleEvents);
+        },
+
+        initialize: function(options){
+            BaseView.prototype.initialize.call(this);
+            this.containerName = options.containerName;
+            this.tplUserURL = options.tplUserURL;
+
+            this.roles = options.roles; // [{key:role_key, name:Human-readable Name}, {key: admin, name: Admin}]
+            this.users = options.users; // [{username: username, email: email, role: role}, ...]
+            this.allow_actions = options.allow_actions;
+            this.current_user_id = options.current_user_id;
+
+            this.initial_role = this.roles[0];
```

Comment:
I didn't realize that the ordering of the roles is important. A comment where the roles are created would be good.